client/src/components/userAuthForm/index.js

```diff
@@ -0,0 +1,225 @@
+import React, {useState} from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+
+import TextField from '@material-ui/core/TextField';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import Select from '@material-ui/core/Select';
+import Button from '@material-ui/core/Button';
+import Snackbar from '@material-ui/core/Snackbar';
+import MuiAlert from '@material-ui/lab/Alert';
+
+function Alert(props) {
+  return <MuiAlert elevation={6} variant="filled" {...props} />
+}
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: 'flex',
+    flexDirection: 'Column',
+    marginTop: "40px"
+  },
+  formField: {
+    width: '80%',
+    marginBottom: '40px',
+  },
+  textField: {
+    marginLeft: theme.spacing(1),
+    marginRight: theme.spacing(1),
+    width: '25ch'
+  },
+  header: {
+    marginTop: '80px'
+  },
+  languageHeader: {
+    marginTop: '20px'
+  },
+  ctaBTN: {
+    color: '#fff',
+    backgroundColor: '#3A8DFF',
+    width: '40%',
+    margin: '0 17.5%',
+    padding: '20px 30px'
+  }
+}));
+
+//
+function validateInput(email, password) {
+  // true means invalid, so our conditions got reversed
+  return {
+    email: email.length === 0,
+    password: password.length === 0
+  };
+}
+
+export default function UserAuthForm({headerText}) {
+  const classes = useStyles();
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [language, setLanguage] = useState('');
+  const [open, setOpen] = useState(false);
+  const [isAlert, setIsAlert] = useState(false);
+
+  const [touched, setTouched] = useState({
+    email: false,
+    password: false
+  });
+
+  const [emailHelperText, setEmailHelperText] = useState('')
+  const [passwordHelperText, setPasswordHelperText] = useState('')
+
+  const handleBlur = (field) => (evt) => {
+    evt.preventDefault();
+    setTouched({...touched, [field]: true})
+  }
+
+  const handleLanguage = (event) => {
+    setLanguage(event.target.value);
+  };
+
+  const handleEmail = (event) => {
+    setEmail(event.target.value);
+    if (validateEmail()) {
+      setEmailHelperText('')
+    }
+  };
+
+  const handlePassword = (event) => {
+    setPassword(event.target.value);
+    if (validatePass()) {
+      setPasswordHelperText('');
+    } 
+  };
+
+  const errors = validateInput(email, password);
+  const isDisabled = Object.keys(errors).some(x => errors[x]);
+
+  function validateEmail() {
+    let regex = /^[^@]+@[^@]+\.[^@]+$/
+    if (regex.test(email) === false) {
+      setEmailHelperText("That is not an email");
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  function validatePass() {
+    if (password.length < 6) {
+      setPasswordHelperText("Password needs to be at least 6 characters");
+      return false;
+    } else {
+      return true;
```

Comment:
```suggestion
     const invalidLength = password.length < 6;
     
     if (invalidLength) setPasswordHelperText("Password needs to be at least 6 characters");
     
     return invalidLength

```

a suggestion on how to reduce num lines of code throughout this page that is full of validity functions :)